system-test/system.ts

```diff
@@ -5,4 +5,122 @@
  * See file LICENSE for detail or copy at https://opensource.org/licenses/MIT
  */
 
-console.warn(`no system tests available 👻`);
+import * as assert from 'assert';
+import * as fs from 'fs';
+import * as gcx from 'gcx';
+import {cloudfunctions_v1, google} from 'googleapis';
+import fetch from 'node-fetch';
+import * as path from 'path';
+import {promisify} from 'util';
+import * as uuid from 'uuid';
+
+const spawn = require('await-spawn');
+const mv = promisify(fs.rename);
+const pkg = require('../../package.json');
+const projectId = process.env.GCLOUD_PROJECT;
+
+if (!projectId) {
+  throw new Error('Please set the `GCLOUD_PROJECT` environment variable.');
+}
+
+let gcf: cloudfunctions_v1.Cloudfunctions;
+const shortPrefix = 'gcloud-tests';
+const fullPrefix = `${shortPrefix}-${uuid.v4().split('-')[0]}`;
+
+describe('gcp metadata', () => {
+  before(async () => {
+    // Clean up any old cloud functions just hanging out
+    gcf = await getGCFClient();
+    await pruneFunctions(false);
+
+    // pack up the gcp-metadata module and copy to the target dir
+    await packModule();
+
+    // deploy the function to GCF
+    await deployApp();
+  });
+
+  it('should access the metadata service on GCF', async () => {
+    const url =
+        `https://us-central1-${projectId}.cloudfunctions.net/${fullPrefix}`;
+    const res = await fetch(url);
```

Comment:
Would this fetch work immediately after a deploy? Did I miss a delay somewhere?

Reply:
`gcx`, which does the deploy here, polls the operation and doesn't resolve the promise until the deployment is done done.